### PR TITLE
feat(1717): latest build

### PR DESCRIPTION
**Pull request by klu909**
## Context

user request a new endpoint to see the most recent build

## Objective

add an end point that returns latest build

## References

https://github.com/screwdriver-cd/screwdriver/issues/1717

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/jobs/latestBuild.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                .then((job) => {
+                    if (!job) {
+                        throw boom.notFound('Job does not exist');
+                    }
+
+                    return job.getLatestBuild({ status });
+                })
+                .then((build) => {
+                    if (Object.getOwnPropertyNames(build).length === 0) {
+                        return reply({});
```
Comment (d2lam) [inline plugins/jobs/latestBuild.js]: I'm also unsure whether to return `{}` or `notFound` here. Want to hear others' opinion on this

Comment (tkyi) [reply plugins/jobs/latestBuild.js]: Not sure as well, but I'm okay with `{}`

Comment (jithin1987) [reply plugins/jobs/latestBuild.js]: I think 404 is semantically correct. With empty object, a client now will have to check not only http status but also look into the return data to see if there is actual result  404 eliminates this problem.

Comment (jithin1987) [reply plugins/jobs/latestBuild.js]: I would say return 404 with body saying `{status: "build does not exist"}` (consistent with responses in other places)

File: plugins/jobs/latestBuild.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            return jobFactory.get(request.params.id)
+                .then((job) => {
+                    if (!job) {
+                        throw boom.notFound('Job does not exist');
+                    }
+
+                    return job.getLatestBuild({ status });
+                })
+                .then((build) => {
+                    if (Object.getOwnPropertyNames(build).length === 0) {
```
Comment (adong) [inline plugins/jobs/latestBuild.js]: `Object.keys()`

Comment (klu909) [reply plugins/jobs/latestBuild.js]: changed done

**Pull request merged**